modules/tensor_mechanics/src/materials/lagrangian/ComputeSimoHughesJ2PlasticityStress.C

```diff
@@ -0,0 +1,194 @@
+
+#include "ComputeSimoHughesJ2PlasticityStress.h"
+
+registerMooseObject("TensorMechanicsApp", ComputeSimoHughesJ2PlasticityStress);
+
+InputParameters
+ComputeSimoHughesJ2PlasticityStress::validParams()
+{
+  InputParameters params = DerivativeMaterialInterface<ComputeLagrangianStressPK1>::validParams();
+  params += SingleVariableReturnMappingSolution::validParams();
+  params.addClassDescription("The Simo-Hughes style J2 plasticity.");
+  params.addParam<MaterialPropertyName>(
+      "elasticity_tensor", "elasticity_tensor", "The name of the elasticity tensor.");
+  params.addRequiredParam<MaterialName>("flow_stress_material",
+                                        "The material defining the flow stress");
+  return params;
+}
+
+ComputeSimoHughesJ2PlasticityStress::ComputeSimoHughesJ2PlasticityStress(
+    const InputParameters & parameters)
+  : DerivativeMaterialInterface<ComputeLagrangianStressPK1>(parameters),
+    GuaranteeConsumer(this),
+    SingleVariableReturnMappingSolution(parameters),
+    _elasticity_tensor_name(_base_name + getParam<MaterialPropertyName>("elasticity_tensor")),
+    _elasticity_tensor(getMaterialProperty<RankFourTensor>(_elasticity_tensor_name)),
+    _F_old(getMaterialPropertyOld<RankTwoTensor>(_base_name + "deformation_gradient")),
+    _ep_name(_base_name + "effective_plastic_strain"),
+    _ep(declareProperty<Real>(_ep_name)),
+    _ep_old(getMaterialPropertyOldByName<Real>(_ep_name)),
+    _be(declareProperty<RankTwoTensor>(_base_name +
+                                       "volume_preserving_elastic_left_cauchy_green_strain")),
+    _be_old(getMaterialPropertyOldByName<RankTwoTensor>(
+        _base_name + "volume_preserving_elastic_left_cauchy_green_strain")),
+    _Np(declareProperty<RankTwoTensor>(_base_name + "flow_direction")),
+    _flow_stress_material(nullptr),
+    _flow_stress_name(_base_name + "flow_stress"),
+    _H(getMaterialPropertyByName<Real>(_flow_stress_name)),
+    _dH(getDefaultMaterialPropertyByName<Real, false>(
+        derivativePropertyName(_flow_stress_name, {_ep_name}))),
+    _d2H(getDefaultMaterialPropertyByName<Real, false>(
+        derivativePropertyName(_flow_stress_name, {_ep_name, _ep_name})))
+{
+}
+
+void
+ComputeSimoHughesJ2PlasticityStress::initialSetup()
+{
+  _flow_stress_material = &getMaterial("flow_stress_material");
+
+  // Enforce isotropic elastic tensor
+  if (!hasGuaranteedMaterialProperty(_elasticity_tensor_name, Guarantee::ISOTROPIC))
+    mooseError("ComputeSimoHughesJ2PlasticityStress requires an isotropic elasticity tensor");
+}
+
+void
+ComputeSimoHughesJ2PlasticityStress::initQpStatefulProperties()
+{
+  ComputeLagrangianStressPK1::initQpStatefulProperties();
+  _be[_qp].setToIdentity();
+  _ep[_qp] = 0;
+}
+
+void
+ComputeSimoHughesJ2PlasticityStress::computeQpPK1Stress()
+{
+  usingTensorIndices(i, j, k, l, m);
+  const Real G = ElasticityTensorTools::getIsotropicShearModulus(_elasticity_tensor[_qp]);
```

Comment:
Oh, okay.  I guess this is fine for the way things are setup now.  If we need to add modularity on the elastic model we can do so later.  Ignore my elasticity tensor comments from above.